.github/workflows/pytest.yml

```diff
@@ -0,0 +1,29 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
```

Comment:
maybe rename the file to code_quality.yml since the intention is to do more than just run pytest.
The linting mentioned is not yet added as a step
I would also add a code coverage check

Reply:
Renamed and coverage added